metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -908,8 +908,11 @@ def _is_allowed_project(self, project: TableauProject) -> bool:
         return is_allowed
 
     def _is_denied_project(self, project: TableauProject) -> bool:
-        # Either project name or project path should exist in deny
-        for deny_pattern in self.config.project_pattern.deny:
+        # Either project_pattern or project_path_pattern is set
+        deny_patterns: List[str] = self.config.project_pattern.deny
+        deny_patterns.extend(self.config.project_path_pattern.deny)
+
+        for deny_pattern in deny_patterns:
             # Either name or project path is denied
             if re.match(
                 deny_pattern, project.name, self.config.project_pattern.regex_flags
```

Comment:
Why are we doing this? We should match the project name against `self.config.project_pattern` and the project path against `self.config.project_path_pattern`, but matching both values against both patterns does not make sense

Are there some backwards compat concerns here? If so, what are they?

Reply:
Simplified the code